UnityProject/Assets/Scripts/Items/Others/NullRod.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Mirror;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Lets null rod and its transformations change from one to another. Can only be used twice before losing the
+/// ability to transform.
+/// </summary>
+public class NullRod : NetworkBehaviour, IInteractable<HandActivate>, IServerSpawn, IServerDespawn
+{
+
+	//Number of times the object can transform. Decreases by one everytime a transformation occurs.
+	[SyncVar(hook=nameof(SyncTransformTimes))]
+	private int transformTimes;
+
+	//Public reference to times left.
+	public int TransformTimes => transformTimes;
+
+	//Changes transformTimes syncvar.
+	private void SyncTransformTimes(int oldTimesLeft, int timesLeft)
+	{
+			this.transformTimes = timesLeft;
+	}
+
+	public void OnSpawnServer()
+	{
+		SyncTransformTimes(transformTimes, 2);
+
+	}
+
+	public override void OnStartClient()
+	{
+		SyncTransformTimes(transformTimes, this.TransformTimes);
+		base.OnStartClient();
+	}
+
+	[Server]
+	public void ServerAdjustTransformTimes(int newTimes)
+	{
+		SyncTransformTimes(transformTimes, newTimes);
+	}
+
+	public void OnSpawnServer(SpawnInfo info)
+	{
+		SyncTransformTimes(transformTimes, 2);
+	}
+
+	public bool WillInteract(HandActivate interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
```

Comment:
Not important, will accept the review anyway.
Could be simplified:
return DefaultWillInteract.Default(interaction, side);

Reply:
Noted!